### DISCUSSION TITLE
When will react-router be fully compatible with react19?

### DISCUSSION THREAD

**Discussion by chinfeng** [remix-run/react-router#12608] [Proposals]
In which version do you plan to achieve full compatibility with react19?

**Comment by sergiodxa** [comment]
React Router v7 already works with React 19, if you mean with RSC that's in development AFAIK, but the team doesn't give an ETA, it will be done once it's done

**Comment by chinfeng** [comment]
According to #12498, rr7 is not  yet compatible with react19.

**Reply by sergiodxa** [reply]
I upgraded to React 19 without issues, the issue may be with SPA mode.
